src/he/msg_handlers.c

```diff
@@ -243,6 +243,24 @@ he_return_code_t he_handle_msg_auth(he_conn_t *conn, uint8_t *packet, int length
         auth_res = HE_ERR_ACCESS_DENIED_NO_AUTH_USERPASS_HANDLER;
       }
       break;
+    case HE_AUTH_TYPE_TOKEN:
+      if(conn->auth_token_cb) {
+        // Check the packet is big enough
+        if(length <= sizeof(he_msg_auth_token_t)) {
```

Comment:
I just wanted to be sure: why is it `<=` and not `<` here?

Reply:
See the definition of `he_msg_auth_token_t`:

```c

typedef struct he_msg_auth_token {
  he_msg_auth_hdr_t header;
  uint16_t token_length;
  uint8_t token[];
} he_msg_auth_token_t;
```

The field `uint8_t token[];` takes 0 byte. So if the `length` is equal to the size of `he_msg_auth_token`, there won't be any room for the actual token data. Therefore the message is too small for a valid auth token message.

Reply:
Oh right, the "Flexible Array Member" feature of C99. Gotcha!  

For clarity, we can more explicitly describe this as `//// Check the packet has at least some token data beyond the fixed struct members` in the future.